src/views/compute/wizard/TemplateIsoSelection.vue

```diff
@@ -232,6 +236,60 @@ export default {
     },
     changeFilterType (value) {
       this.filterType = value
+    },
+    getTemplatesByKeyword (keyword) {
+      let apiCommand = ''
+      let apiCommandResponse = ''
+      this.filteredItems = []
+      const promises = []
+      const templates = []
+      const params = {}
+      const templateFilter = [
+        'featured',
+        'community',
+        'selfexecutable',
+        'sharedexecutable'
+      ]
+      // Fetch templates or isos depending on active tab
+      params.zoneid = _.get(this.zone, 'id')
+      params.keyword = keyword
+      params.showunique = true
+      if (this.inputDecorator === 'templateid') {
+        apiCommand = 'listTemplates'
+        apiCommandResponse = 'listtemplatesresponse.template'
+      } else {
+        apiCommand = 'listIsos'
+        apiCommandResponse = 'listisosresponse.iso'
+        params.bootable = 'true'
+      }
+      this.isLoading = true
+      // Fetch templates/isos for all filters
+      templateFilter.forEach(filter => {
+        params.templatefilter = filter
+        promises.push(this.fetchTemplatesIsos(apiCommand, keyword, params))
+      })
+
+      Promise.all(promises).then(response => {
+        response.forEach((resItem) => {
+          const concatTemplates = _.concat(templates, _.get(resItem, apiCommandResponse, []))
```

Comment:
@rhtyd @ravening Keeping it this way to not introduce too many changes, but since we're getting the API response per filter, we can set the datasource here itself. Would also need changes in DeployVM

Reply:
Also zoneid doesn't get passed, need to address that too but preferably in another PR where we move the search functionality to DeployVM like other view do

Reply:
yeah zoneid is sometimes a crucial parameter. code LGTM apart from the above mentioned missing stuffs